cpp/test/sg/fnv_hash_test.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuml/fil/fnv_hash.h>
+#include <gtest/gtest.h>
+#include <raft/error.hpp>
+
+typedef std::tuple<std::vector<char>, unsigned long long, unsigned long>
+  fnv_vec_t;
+
+class FNVHashTest : public testing::TestWithParam<fnv_vec_t> {
+ protected:
+  void SetUp() override { std::tie(input, correct64, correct32) = GetParam(); }
+
+  void check() {
+    unsigned long long real64 =
+      fowler_noll_vo_fingerprint64(input.begin(), input.end());
+    ASSERT(real64 == correct64, "Wrong hash computed");
+    unsigned long real32 =
+      fowler_noll_vo_fingerprint64_32(input.begin(), input.end());
+    ASSERT(real32 == correct32, "Wrong xor-folded hash computed");
+  }
+
+  // parameters
+  std::vector<char> input;
+  unsigned long long correct64;
+  unsigned long correct32;
+};
+
+std::vector<fnv_vec_t> fnv_vecs = {
+  {{}, 14695981039346656037ull, 0xcbf29ce4 ^ 0x84222325},  // test #0
+  {{0}, 0xaf63bd4c8601b7df, 0xaf63bd4c ^ 0x8601b7df},
+  {{1}, 0xaf63bd4c8601b7de, 0xaf63bd4c ^ 0x8601b7de},
+  {{2}, 0xaf63bd4c8601b7dd, 0xaf63bd4c ^ 0x8601b7dd},
+  {{3}, 0xaf63bd4c8601b7dc, 0xaf63bd4c ^ 0x8601b7dc},
+  {{1, 2}, 0x08328707b4eb6e38, 0x08328707 ^ 0xb4eb6e38},  // test #5
+  {{2, 1}, 0x08328607b4eb6c86, 0x08328607 ^ 0xb4eb6c86},
+  {{1, 2, 3}, 0xd949aa186c0c492b, 0xd949aa18 ^ 0x6c0c492b},
```

Comment:
Aren't there any other test inputs available?

Reply:
Yes, there are, but those do not exercise the properties that FIL needs. If you want to build an FNV library to cater to all kinds of applications, we can start a design doc and discuss the scope there?